v2/cf-nodejs-logging-support/src/lib/config/config.ts

```diff
@@ -163,12 +169,67 @@ export default class Config {
                     field._meta.isEnabled = false;
                 }
 
+                // check if cache field
+                if (!Array.isArray(field.source)) {
+                    if (["static", "env"].includes(field.source.type)) {
+                        field._meta.isCache = true;
+                    }
+                } else {
+                    // if Sources[] then only check first source
+                    if (["static", "env"].includes(field.source[0].type)) {
+                        field._meta.isCache = true;
+                    }
+                }
+
+                if (field.output?.includes('msg-log')) {
+                    this.addToList(this.msgFields, field);
+
+                }
+
+                if (field.output?.includes('req-log')) {
+                    this.addToList(this.reqFields, field);
+
+                }
+
+                // check if context field, if true, then save field in list
+                if (field.output?.includes("msg-log") && field.output?.includes("req-log")) {
+                    if (Array.isArray(field.source)) {
+                        const sources = field.source as Source[];
+                        for (let index = 0; index < sources.length; index++) {
+                            const source = sources[index];
+                            if (["req-header", "req-object"].includes((source.type))) {
+                                field._meta.isContext = true;
+                                this.addToList(this.contextFields, field);
+                                break;
+                            }
+                        }
+                    } else {
+                        if (["req-header", "req-object"].includes((field.source.type))) {
+                            field._meta.isContext = true;
+                            this.addToList(this.contextFields, field);
+                        }
+                    }
+
+                    if (["correlation_id", "tenant_id"].includes(field.name)) {
```

Comment:
Actual field names should not be referenced in code. 

We should avoid over-engineering the concept of context fields. If we cannot come up with a proper condition based on the existing configuration approach we have to add a new config value to indicate that a field is a context field. The important part is that context fields are only present if a log is a request log or a message log getting logged within the context of a request. However, it would be a valid use case to have a field that is not a context field, but present for all request and all message logs.

Reply:
Makes sense. I added a new property `isContext` in the configuration to directly identify context fields without over-engineering.